### PR TITLE
RTS-1466: Reduce the limits on warn_object_size and max_object_size

### PR DESCRIPTION
**Pull request by hazen**
https://bashoeng.atlassian.net/browse/RTS-1466

Basically this is a port of https://github.com/basho/riak_kv/blob/develop/src/riak_kv_vnode.erl#L2338-L2353 to write-once.

Test: https://github.com/basho/riak_test/pull/1218

### CODE REVIEW

**Comment by korry8911** [comment]
+1 looks fine to me

**Comment by korry8911** [comment]
what about https://github.com/basho/riak_kv/blob/79473edc4ef6f90b29cf7394186c01ab045ad69d/test/riak_kv_schema_tests.erl#L102

https://github.com/basho/riak_kv/blob/79473edc4ef6f90b29cf7394186c01ab045ad69d/test/riak_kv_schema_tests.erl#L103

https://github.com/basho/riak_kv/blob/79473edc4ef6f90b29cf7394186c01ab045ad69d/test/riak_kv_schema_tests.erl#L159

https://github.com/basho/riak_kv/blob/79473edc4ef6f90b29cf7394186c01ab045ad69d/test/riak_kv_schema_tests.erl#L160
@javajolt

**Comment by hazen** [comment]
Those look to simply be configuration test values. They only live inside the test, so they are probably good to change them as well, however.

**Comment by gordonguthrie** [comment]
+1

**Comment by andytill** [comment]
Is this to check Riak TS writes? From the `riak_kv_vnode.erl` code it looks like it only affects normal puts.

**Comment by hazen** [comment]
Sounds like we need to review this requirement

**Comment by hazen** [comment]
@andytill Where are write once limits imposed?

**Comment by andytill** [comment]
I can't see where the limits are used at all, by looking for that configuration key. I haven't investigated thoroughly, just did a find all on the key string.

**Comment by ph07** [comment]
Discussed with @angrycub - we will have lower thresholds for Riak TS 1.5 as defined in the requirements document. For 2017 roadmap, we will discuss how to make it more flexible. We'll probably need several global thresholds (e.g. KV buckets, TS tables, object storage) and also will check what it takes to have limits per bucket type.

File: src/riak_kv_w1c_worker.erl [hunk 1]
```diff
@@ -222,34 +231,54 @@ async_put_replies(ReqIdTuples, Options) ->
...
-                       S#state{proxies=UpdProxies};
-                   {_, S} ->
-                       reply(From, ReqId, {error, request_id_already_defined}),
-                       S
-    end,
+handle_cast({batch_put, Bucket, BucketProps, EncodedVals, ReqId, PK, Preflist, Rec}, State) ->
+    #rec{from=From, obj_sizes=BinSizes} = Rec,
+    #state{proxies=Proxies, max_object_size=MaxSize, warn_object_size=WarnSize} = State,
+    %% Report or fail on large objects
+    NewState = case are_object_sizes_valid(Bucket, PK, BinSizes, MaxSize, WarnSize) of
```
Comment (macintux) [inline src/riak_kv_w1c_worker.erl:267]: This nested `case` statement falls afoul of our preferred Erlang style. I don't think it warrants a rewrite, but this function is getting a bit clunky.

File: src/riak_kv_w1c_worker.erl [hunk 2]
```diff
@@ -542,3 +571,33 @@ chash_key(Bucket, {PartitionKey, _LocalKey}, BucketProps) ->
...
+are_object_sizes_valid(Bucket, Key, BinSizes, MaxSize, WarnSize) ->
+    TooBig = lists:filter(fun(S) ->
+                              S > MaxSize
+                          end, BinSizes),
+    case TooBig of
+        [] ->
+            Warning = lists:filter(fun(S) ->
+                                       S > WarnSize
+                                   end, BinSizes),
+            case Warning of
```
Comment (macintux) [inline src/riak_kv_w1c_worker.erl:591]: Ditto

**Comment by macintux** [comment]
After some quick experimentation I can't think of a good way to clean this up, so never mind my suggestions above.

Test works, logging matches, code looks like it's doing what's intended.

+1

**Comment by davidx** [comment]
thumbot retry

**Comment by paegun** [comment]
+1

**Comment by davidx** [comment]
thumbot retry

**Comment by paegun** [comment]
thumbot retry

**Comment by paegun** [comment]
thumbot retry

**Comment by korry8911** [comment]
+1

**Pull request merged**